debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -60,8 +60,8 @@
 
     private static final String STATEMENTS_PLACEHOLDER = "#";
     private static final String GET_MAX_LSN = "SELECT sys.fn_cdc_get_max_lsn()";
-
-    private static final String GET_MAX_LSN_SKIP_LOW_ACTIVTY = "SELECT (SELECT MAX(start_lsn) FROM cdc.lsn_time_mapping), (SELECT MAX(start_lsn) FROM cdc.lsn_time_mapping WHERE tran_id <> 0x00)";
+    private static final String GET_MAX_TRANSACTION_LSN = "SELECT MAX(start_lsn) FROM cdc.lsn_time_mapping WHERE tran_id <> 0x00";
+    private static final String GET_NTH_TRANSACTION_LSN = "SELECT MAX(start_lsn) FROM (SELECT TOP (?) start_lsn FROM cdc.lsn_time_mapping WHERE tran_id <> 0x00# ORDER BY start_lsn) as next_lsns";
```

Comment:
The execution plan of the new query looks like the following:
```
+-----------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+--------+------------------+-----------+--------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------------------------------------------+-----------------------------------------------+----------+-----------+----------+--------------------+------------------------------------------------+---------------------------------------------+-----------------------------------------+----------+-------+---+
| StmtText                                                                                                        | StmtId                                                                          | NodeId                                                                                                                          | Parent | PhysicalOp       | LogicalOp | Argument                 | DefinedValues                                                 | EstimateRows                                                                | EstimateIO                                                                      | EstimateCPU                                                                                                                     | AvgRowSize                                     | TotalSubtreeCost                              | OutputList                                    | Warnings | Type      | Parallel | EstimateExecutions |                                                |                                             |                                         |          |       |   |
+-----------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+--------+------------------+-----------+--------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------------------------------------------+-----------------------------------------------+----------+-----------+----------+--------------------+------------------------------------------------+---------------------------------------------+-----------------------------------------+----------+-------+---+
| SELECT MAX(start_lsn)                                                                                           |                                                                                 |                                                                                                                                 |        |                  |           |                          |                                                               |                                                                             |                                                                                 |                                                                                                                                 |                                                |                                               |                                               |          |           |          |                    |                                                |                                             |                                         |          |       |   |
| FROM (SELECT TOP (10000) start_lsn                                                                              |                                                                                 |                                                                                                                                 |        |                  |           |                          |                                                               |                                                                             |                                                                                 |                                                                                                                                 |                                                |                                               |                                               |          |           |          |                    |                                                |                                             |                                         |          |       |   |
| FROM cdc.lsn_time_mapping                                                                                       |                                                                                 |                                                                                                                                 |        |                  |           |                          |                                                               |                                                                             |                                                                                 |                                                                                                                                 |                                                |                                               |                                               |          |           |          |                    |                                                |                                             |                                         |          |       |   |
| WHERE start_lsn >= 0x0000002600000C58001A AND tran_id <> 0x00                                                   |                                                                                 |                                                                                                                                 |        |                  |           |                          |                                                               |                                                                             |                                                                                 |                                                                                                                                 |                                                |                                               |                                               |          |           |          |                    |                                                |                                             |                                         |          |       |   |
| ORDER BY start_lsn) as next_lsns"                                                                               | 1                                                                               | 1                                                                                                                               | 0      |                  |           | 1                        |                                                               | 1                                                                           |                                                                                 |                                                                                                                                 |                                                | 0.0032874597                                  |                                               |          | SELECT    | false    |                    |                                                |                                             |                                         |          |       |   |
|     |--Stream Aggregate(DEFINE:([Expr1002]=MAX([testDB].[cdc].[lsn_time_mapping].[start_lsn])))                 | 1                                                                               | 2                                                                                                                               | 1      | Stream Aggregate | Aggregate |                          | [Expr1002]=MAX([testDB].[cdc].[lsn_time_mapping].[start_lsn]) | 1                                                                           | 0                                                                               | 0.0000013569723                                                                                                                 | 17                                             | 0.0032874597                                  | [Expr1002]                                    |          | PLAN_ROW  | false    | 1                  |                                                |                                             |                                         |          |       |   |
|         |--Top(TOP EXPRESSION:((10000)))                                                                        | 1                                                                               | 3                                                                                                                               | 2      | Top              | Top       | TOP EXPRESSION:((10000)) |                                                               | 1.4282871                                                                   | 0                                                                               | 0.00000014282872                                                                                                                | 17                                             | 0.0032861028                                  | [testDB].[cdc].[lsn_time_mapping].[start_lsn] |          | PLAN_ROW  | false    | 1                  |                                                |                                             |                                         |          |       |   |
|             |--Clustered Index Seek(OBJECT:([testDB].[cdc].[lsn_time_mapping].[lsn_time_mapping_clustered_idx]) |  SEEK:([testDB].[cdc].[lsn_time_mapping].[start_lsn] >= 0x0000002600000C58001A) |   WHERE:([testDB].[cdc].[lsn_time_mapping].[tran_id]<0x00 OR [testDB].[cdc].[lsn_time_mapping].[tran_id]>0x00) ORDERED FORWARD) | 1      | 4                | 3         | Clustered Index Seek     | Clustered Index Seek                                          | OBJECT:([testDB].[cdc].[lsn_time_mapping].[lsn_time_mapping_clustered_idx]) |  SEEK:([testDB].[cdc].[lsn_time_mapping].[start_lsn] >= 0x0000002600000C58001A) |   WHERE:([testDB].[cdc].[lsn_time_mapping].[tran_id]<0x00 OR [testDB].[cdc].[lsn_time_mapping].[tran_id]>0x00) ORDERED FORWARD" | "[testDB].[cdc].[lsn_time_mapping].[start_lsn] |  [testDB].[cdc].[lsn_time_mapping].[tran_id]" | 1.4282871                                     | 0.003125 | 0.0001592 | 24       | 0.0032842          | "[testDB].[cdc].[lsn_time_mapping].[start_lsn] | [testDB].[cdc].[lsn_time_mapping].[tran_id] | NO STATS:([lsn_time_mapping].[tran_id]) | PLAN_ROW | false | 1 |
+-----------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+--------+------------------+-----------+--------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------+---------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------------------------------------------+-----------------------------------------------+----------+-----------+----------+--------------------+------------------------------------------------+---------------------------------------------+-----------------------------------------+----------+-------+---+
```
There doesn't seem to be any red flags.